include/range/v3/view/adjacent_remove_if.hpp

```diff
@@ -38,65 +40,108 @@ namespace ranges
                 adjacent_remove_if_view<Rng, Pred>,
                 Rng,
                 is_finite<Rng>::value ? finite : range_cardinality<Rng>::value>
+          , private box<semiregular_t<Pred>, adjacent_remove_if_view<Rng, Pred>>
         {
+            adjacent_remove_if_view() = default;
+            constexpr adjacent_remove_if_view(Rng rng, Pred pred)
+                noexcept(
+                    std::is_nothrow_constructible<
+                        typename adjacent_remove_if_view::view_adaptor, Rng>::value &&
+                    std::is_nothrow_constructible<
+                        typename adjacent_remove_if_view::box, Pred>::value)
+              : adjacent_remove_if_view::view_adaptor{detail::move(rng)}
+              , adjacent_remove_if_view::box(detail::move(pred))
+            {}
         private:
             friend range_access;
-            semiregular_t<Pred> pred_;
-            detail::non_propagating_cache<iterator_t<Rng>> begin_;
 
             struct adaptor : adaptor_base
             {
             private:
                 adjacent_remove_if_view *rng_;
-                void satisfy(iterator_t<Rng> &it) const
-                {
-                    auto const end = ranges::end(rng_->mutable_base());
-                    auto &pred = rng_->pred_;
-                    if(it == end)
-                        return;
-                    auto next = it;
-                    for(; ++next != end; it = next)
-                        if(!invoke(pred, *it, *next))
-                            return;
-                }
             public:
                 adaptor() = default;
-                adaptor(adjacent_remove_if_view &rng)
+                constexpr adaptor(adjacent_remove_if_view &rng) noexcept
                   : rng_(&rng)
                 {}
-                iterator_t<Rng> begin(adjacent_remove_if_view &) const
+                RANGES_CXX14_CONSTEXPR static iterator_t<Rng> begin(adjacent_remove_if_view &rng)
+                    noexcept(std::is_nothrow_copy_constructible<iterator_t<Rng>>::value)
+                {
+                    return *rng.begin_;
+                }
+                RANGES_CXX14_CONSTEXPR void next(iterator_t<Rng> &it) const
+                    noexcept(noexcept((void)(it != ranges::end(std::declval<Rng &>())),
+                        std::declval<adjacent_remove_if_view &>().satisfy_forward(++it)))
```

Comment:
It's egregiously silly to have exception specifiers that include calls to `satisfy_(forward|reverse)` which is always `noexcept(true)`.